### PR TITLE
mbedtls: update Kconfig and mbedtls_config.h

### PR DESCRIPTION
**Pull request by ThePassionate**
update Kconfig and mbedtls_config.h for version v3.4.0
(1)MBEDTLS_SSL_MAX_CONTENT_LEN is splited into  MBEDTLS_SSL_IN_CONTENT_LEN and MBEDTLS_SSL_OUT_CONTENT_LEN 
(2) MBEDTLS_REMOVE_ARC4_CIPHERSUITES and MBEDTLS_CERTS_C is remved in v3.4.0
(3) update some configs status

### CODE REVIEW

File: crypto/mbedtls/Kconfig [hunk 1]
```diff
@@ -6,6 +6,7 @@
 menuconfig CRYPTO_MBEDTLS
 	bool "Mbed TLS Cryptography Library"
 	default n
+	select DEV_URANDOM
```
Comment (pkarashchenko) [inline crypto/mbedtls/Kconfig:9]: I'm not sure about this change. Could you please explain?
I think we should explore usage of `HAVE_GETRANDOM` from MbedTLS as NuttX has `getrandom` implementation.

Comment (ThePassionate) [reply crypto/mbedtls/Kconfig:9]: I agree with your point of view， using the system's `getrandom` is better than obtaining it from `urandom`

**Comment by ThePassionate** [comment]
> update Kconfig and mbedtls_config.h for version v3.4.0 (1)MBEDTLS_SSL_MAX_CONTENT_LEN is splited into MBEDTLS_SSL_IN_CONTENT_LEN and MBEDTLS_SSL_OUT_CONTENT_LEN (2) MBEDTLS_REMOVE_ARC4_CIPHERSUITES and MBEDTLS_CERTS_C is remved in v3.4.0 (3) update some configs status Signed-off-by: makejian [makejian@xiaomi.com](mailto:makejian@xiaomi.com)

depends on 

> update Kconfig and mbedtls_config.h for version v3.4.0 (1)MBEDTLS_SSL_MAX_CONTENT_LEN is splited into MBEDTLS_SSL_IN_CONTENT_LEN and MBEDTLS_SSL_OUT_CONTENT_LEN (2) MBEDTLS_REMOVE_ARC4_CIPHERSUITES and MBEDTLS_CERTS_C is remved in v3.4.0 (3) update some configs status Signed-off-by: makejian [makejian@xiaomi.com](mailto:makejian@xiaomi.com)

depends on commit https://github.com/apache/nuttx/pull/10407

File: crypto/mbedtls/Kconfig [hunk 2]
```diff
@@ -47,10 +53,6 @@ config MBEDTLS_AES_ROM_TABLES
 	bool "Store the AES tables in ROM."
 	default n
 
-config MBEDTLS_REMOVE_ARC4_CIPHERSUITES
```
Comment (xiaoxiang781216) [inline crypto/mbedtls/Kconfig:50]: why remove

File: crypto/mbedtls/Kconfig [hunk 3]
```diff
@@ -61,68 +63,64 @@ config MBEDTLS_ECP_RESTARTABLE
...
 config MBEDTLS_BLOWFISH_C
 	bool "Enable the Blowfish block cipher."
-	default n
+	default y
 
 config MBEDTLS_CAMELLIA_C
 	bool "Enable the Camellia block cipher."
-	default n
-
-config MBEDTLS_CERTS_C
```
Comment (xiaoxiang781216) [inline crypto/mbedtls/Kconfig:78]: why remove

File: crypto/mbedtls/Kconfig [hunk 4]
```diff
@@ -61,68 +63,64 @@ config MBEDTLS_ECP_RESTARTABLE
...
 
 config MBEDTLS_SSL_DTLS_CLIENT_PORT_REUSE
 	bool "Enable server-side support for clients that reconnect from the same port."
-	default n
+	default y
 
 config MBEDTLS_BLOWFISH_C
 	bool "Enable the Blowfish block cipher."
-	default n
+	default y
```
Comment (xiaoxiang781216) [inline crypto/mbedtls/Kconfig:74]: why enable blowfish by the default

File: crypto/mbedtls/Kconfig [hunk 5]
```diff
@@ -61,68 +63,64 @@ config MBEDTLS_ECP_RESTARTABLE
...
-	default n
+	default y if !MBEDTLS_AES_ALT
 
 config MBEDTLS_TIMING_C
 	bool "Enable the semi-portable timing interface."
 	default y
 
 config MBEDTLS_SSL_KEEP_PEER_CERTIFICATE
 	bool "Enable the availability of the API mbedtls_ssl_get_peer_cert() giving access to the peer's certificate after completion of the handshake."
-	default n
```
Comment (xiaoxiang781216) [inline crypto/mbedtls/Kconfig:92]: why change the default

File: crypto/mbedtls/Kconfig (hunk 2 above)
Comment (ThePassionate) [reply crypto/mbedtls/Kconfig:50]: removed in v3.4.0 mbedtls_config.h

File: crypto/mbedtls/Kconfig (hunk 3 above)
Comment (ThePassionate) [reply crypto/mbedtls/Kconfig:78]: removed in v3.4.0 mbedtls_config.h has been removed in v3.4.0 mbedtls_config.h

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**